### ISSUE TITLE
Internal Error

### ISSUE THREAD

**Issue by GoogleCodeExporter** [peteaky/android-xbmcremote#66]
```
What steps will reproduce the problem?
1. Install App
2. Wait for settings popup
3. Click [close]
4. Popup "Internal Error" appears.

What is the expected output? What do you see instead?

Internal Error - Wrong data from HTTP API; expected 'OK', got ''.

What version of the product are you using? On what operating system?
0.5.11 and 0.5.13
```

Original issue reported on code.google.com by `bolin.andreas@gmail.com` on 4 Nov 2009 at 8:28

**Comment by GoogleCodeExporter** [comment]
```
More info:
Another problem that seems to be related is that when I'm done with the 
settings and
hit back, XBMC displays a message that the app is connected before I have 
pressed
'connect' in the app. So maybe thats why issue 66 appears?
Cheers
```

Original comment by `bolin.andreas@gmail.com` on 4 Nov 2009 at 8:52

**Comment by GoogleCodeExporter** [comment]
Original comment by `phree...@gmail.com` on 5 Nov 2009 at 12:15
- Changed state: **Accepted**
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
11-05 11:47:46.915: WARN/System.err(1354): 
org.xbmc.httpapi.WrongDataFormatException:
Wrong data format, expected 'OK', got ''.
11-05 11:47:46.921: WARN/System.err(1354):     at
org.xbmc.httpapi.Connection.assertBoolean(Connection.java:285)
11-05 11:47:46.921: WARN/System.err(1354):     at
org.xbmc.httpapi.Connection.setResponseFormat(Connection.java:294)
11-05 11:47:46.921: WARN/System.err(1354):     at
org.xbmc.httpapi.Connection.<init>(Connection.java:97)
11-05 11:47:46.921: WARN/System.err(1354):     at
org.xbmc.httpapi.HttpClient.<init>(HttpClient.java:118)
11-05 11:47:46.921: WARN/System.err(1354):     at
org.xbmc.httpapi.HttpClient.<init>(HttpClient.java:95)
11-05 11:47:46.921: WARN/System.err(1354):     at
org.xbmc.android.util.ConnectionManager.getHttpClient(ConnectionManager.java:164
)
11-05 11:47:46.921: WARN/System.err(1354):     at
org.xbmc.android.backend.httpapi.Wrapper.info(Wrapper.java:55)
11-05 11:47:46.921: WARN/System.err(1354):     at
org.xbmc.android.backend.httpapi.InfoWrapper$1.run(InfoWrapper.java:46)
11-05 11:47:46.930: WARN/System.err(1354):     at
android.os.Handler.handleCallback(Handler.java:587)
11-05 11:47:46.930: WARN/System.err(1354):     at
android.os.Handler.dispatchMessage(Handler.java:92)
11-05 11:47:46.930: WARN/System.err(1354):     at 
android.os.Looper.loop(Looper.java:123)
11-05 11:47:46.930: WARN/System.err(1354):     at
org.xbmc.android.backend.httpapi.HttpApiThread.run(HttpApiThread.java:69)

```

Original comment by `phree...@gmail.com` on 5 Nov 2009 at 10:53

**Comment by GoogleCodeExporter** [comment]
```
Seems like socket timeouts (and the like) gets "handled" several times:
Connection.query() sends exceptions to errorhandler AND returns an empty string.
The empty string then triggers the WrongDataFormatException in 
assertBoolean()/setResponseFormat()
Needs cleanup.
Maybe the NoSettings check/Exception-throwing should be done in 
ConnectionManager?

```

Original comment by `mukkenb...@gmail.com` on 5 Nov 2009 at 11:02

**Comment by GoogleCodeExporter** [comment]
```
Definitely needs cleanup. Will look at it soon.
```

Original comment by `phree...@gmail.com` on 6 Nov 2009 at 12:16

**Comment by GoogleCodeExporter** [comment]
```
Fixed in v0.6.0
We know there are still potential issues if communication breaks down 
in-session, so 
feel free to report any of those that you might encounter!

```

Original comment by `mukkenb...@gmail.com` on 11 Nov 2009 at 6:41
- Changed state: **Fixed**
